server/src-lib/Hasura/RQL/Types/ScheduledTrigger.hs

```diff
@@ -0,0 +1,103 @@
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+
+module Hasura.RQL.Types.ScheduledTrigger
+  ( ScheduleType(..)
+  , CreateScheduledTrigger(..)
+  , RetryConfST(..)
+  , formatTime'
+  ) where
+
+import           Data.Time.Clock
+import           Data.Time.Clock.Units
+import           Data.Time.Format
+import           Data.Aeson
+import           Data.Aeson.Casing
+import           Data.Aeson.TH
+import           Hasura.Prelude
+import           System.Cron.Types
+import           System.Cron.Parser
+import           Hasura.Incremental
+
+import qualified Data.Text                     as T
+import qualified Data.Aeson                    as J
+import qualified Hasura.RQL.Types.EventTrigger as ET
+
+data RetryConfST
+  = RetryConfST
+  { rcstNumRetries  :: !Int
+  , rcstIntervalSec :: !Seconds
+  , rcstTimeoutSec  :: !Seconds
+  , rcstTolerance   :: !NominalDiffTime
+  } deriving (Show, Eq, Generic)
+
+instance NFData RetryConfST
+instance Cacheable RetryConfST
+
+$(deriveJSON (aesonDrop 2 snakeCase){omitNothingFields=True} ''RetryConfST)
+
+defaultRetryConf :: RetryConfST
+defaultRetryConf =
+  RetryConfST
+  { rcstNumRetries = 1
+  , rcstIntervalSec = seconds 10
+  , rcstTimeoutSec = seconds 60
+  , rcstTolerance = 21600 -- 6 hours
```

Comment:
If we update `Data.Time.Clock.Units` to support both `DiffTime` and `NominalDiffTime`, this could be `hours 6`, which would be nicer!

Reply:
Currently, `Seconds` is defined as `newtype Seconds = Seconds { seconds :: DiffTime }`.

Should we define it as `newtype NominalSeconds  = NominalSeconds { NominalSeconds :: NominalDiffTime}` 

or

`newtype Seconds a = Seconds {seconds :: a}`

Reply:
I’d probably do something along these lines:

```haskell
data DurationType = Absolute | Calendar
type family Duration t = r | r -> t where
  Duration 'Absolute = DiffTime
  Duration 'Calendar = NominalDiffTime

newtype Seconds t = Seconds { seconds :: Duration t }
newtype Minutes t = Minutes { minutes :: Duration t }
-- etc.
```